### PR TITLE
Add doc about referencing XAML extensibility

### PR DESCRIPTION
**Pull request by debonte**
Added a new document that covers issues around referencing 
Microsoft.VisualStudio.DesignTools.Extensibility -- how to reference it, which version to use, and shipping multiple copies of your designtools DLL that reference different versions.

### CODE REVIEW

File: documents/referencing-xaml-designer-extensibility.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# Shipping multiple versions of a designtools DLL
+
+You may find that you need to ship more than one version of your designtools DLL, each targeting a different version of Visual Studio. For example, you might have basic design-time support for your controls starting in VS 16.4, but due to the introduction of some new designer extensibility API in 16.9 you might be able to provide a better user experience in 16.9 and later. In that case you would want to ship two designtools DLLs, one referencing version 16.4 of Microsoft.VisualStudio.DesignTools.Extensibility.dll and the other referencing version 16.9.
+
+When the designer looks for a designtools DLL for your control assembly and finds more than one, it will choose the DLL that references the highest version of Microsoft.VisualStudio.DesignTools.Extensibility.dll which is less than or equal to the version of Microsoft.VisualStudio.DesignTools.Extensibility.dll used by the designer (i.e. the version that shipped with the running instance of Visual Studio).
```
Comment (lutzroeder) [inline documents/referencing-xaml-designer-extensibility.md:17]: Same here, file names in italics *Microsoft.VisualStudio.DesignTools.Extensibility.dll*.

[the remark this answers is not in the document]
Comment (debonte) [reply documents/referencing-xaml-designer-extensibility.md]: Fixed

Comment (debonte) [reply documents/referencing-xaml-designer-extensibility.md]: Fixed

**Review by lutzroeder — APPROVED** [review]

File: documents/referencing-xaml-designer-extensibility.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+The following table shows how the XAML designer in VS 17.0 would react when presented with three *designtools.dlls* referencing different versions of *Microsoft.VisualStudio.DesignTools.Extensibility.dll*.
+
+| Referenced version of *Microsoft.VisualStudio.DesignTools.Extensibility.dll* | Will the designer load it in VS 17.0?                             |
+| -----------------------------------------------------------------------------| ----------------------------------------------------------------- |
+| 16.8                                                                         | No. It is compatible with 17.0, but a higher version is available |
+| 16.10                                                                        | Yes. Closest to the designer's version                            |
```
Comment (mgoertz-msft) [inline documents/referencing-xaml-designer-extensibility.md:24]: There's a caveat to the 16.10 Yes answer: unless it references other VS extensibility assemblies due to the Interop breaking change that we may want to link to here too.

Comment (mgoertz-msft) [reply documents/referencing-xaml-designer-extensibility.md:24]: Talked offline. Apparently, it's more complicated than that.

**Review by mgoertz-msft — APPROVED** [review]

**Pull request merged**